SPECS/ipv6calc/ipv6calc.spec

```diff
@@ -0,0 +1,661 @@
+# enable the following for intermediate builds
+#%#define gitcommit d3a4108cb7aeb6f731bb07989f91d8a7f449f0f0
+
+# mod_ipv6calc related
+%{!?_httpd_apxs:    %{expand: %%global _httpd_apxs    %%{_sbindir}/apxs}}
+%{!?_httpd_moddir:  %{expand: %%global _httpd_moddir  %%{_libdir}/httpd/modules}}
+%{!?_httpd_confdir: %{expand: %%global _httpd_confdir %%{_sysconfdir}/httpd/conf.d}}
+# database support (deselectable)
+%if "%{?_without_ip2location:0}%{?!_without_ip2location:1}" == "1"
+%define enable_ip2location 1
+%endif
+%if "%{?_without_geoip:0}%{?!_without_geoip:1}" == "1"
+%define enable_geoip 1
+%endif
+%if "%{?_without_dbip:0}%{?!_without_dbip:1}" == "1"
+%define enable_dbip 1
+%endif
+%if "%{?_without_external:0}%{?!_without_external:1}" == "1"
+%define enable_external 1
+%endif
+# database locations
+%define ip2location_db	%{_datadir}/IP2Location
+%define geoip_db	%{_datadir}/GeoIP
+%define dbip_db		%{_datadir}/DBIP
+%define external_db	%{_datadir}/%{name}/db
+# Berkeley DB selector
+%define require_db4 %(echo "%{dist}" | grep -E -q '^\.el(5|6)$' && echo 1 || echo 0)
+# RPM license macro detector
+%define rpm_license_extra %(echo "%{_defaultlicensedir}" | grep -q defaultlicensedir && echo 0 || echo 1)
+%if 0%{?gitcommit:1}
+%global shortcommit %(c=%{gitcommit}; echo ${c:0:7})
+%define build_timestamp %(date +"%{Y}%{m}%{d}")
+%global gittag .%{build_timestamp}git%{shortcommit}
+%endif
+# shared library support (deselectable)
+%if "%{?_without_shared:0}%{?!_without_shared:1}" == "1"
+%define enable_shared 1
+%endif
+Summary:        IPv6 address format change and calculation utility
+Name:           ipv6calc
+Version:        2.2.0
+Release:        41%{?gittag}%{?dist}
+License:        GPLv2
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            http://www.deepspace6.net/projects/%{name}.html
+Patch0:         ipv6calc-2.2.0-patch-8c7eea58.diff
+BuildRequires:  gcc
+BuildRequires:  openssl-devel
+BuildRequires:  perl
+BuildRequires:  procps-ng
+BuildRequires:  perl(Digest::MD5)
+BuildRequires:  perl(strict)
+BuildRequires:  perl(warnings)
+%if 0%{?gitcommit:1}
+Source:         https://github.com/pbiering/%{name}/archive/%{gitcommit}/%{name}-%{gitcommit}.tar.gz
```

Comment:
Might be worth running the spec linter against the new specs. We tend to keep the `SourceX` tags right under the `URL` one, plust there's a lot of other spec conventions we'd like to follow.

Reply:
I ran the linter but it looks like it got confused by the %if conditionals. After switching it to just the non-gitcommit source the linter puts it under URL correctly.

Reply:
The linter is still being improved. I'll let Thomas know that this is happening when it encounters conditionals.

---
In reply to: [517055486](https://github.com/microsoft/CBL-Mariner/pull/338#discussion_r517055486) [](ancestors = 517055486)